staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -196,14 +204,52 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	hooks := hookConfig.ExternalAdmissionHooks
 	ctx := context.TODO()
 
-	errCh := make(chan error, len(hooks))
+	hookIndexCh := make(chan int, len(hooks))
 	wg := sync.WaitGroup{}
 	wg.Add(len(hooks))
 	for i := range hooks {
+		go func(hook *v1alpha1.ExternalAdmissionHook, i int) {
+			defer wg.Done()
+			if shouldCallHook(hook, attr) {
+				hookIndexCh <- i
+			}
+		}(&hooks[i], i)
+	}
+	wg.Wait()
+	close(hookIndexCh)
+
+	var hookIndice []int
+	for index := range hookIndexCh {
+		hookIndice = append(hookIndice, index)
+	}
+	if len(hookIndice) == 0 {
+		// no matching hooks
+		return nil
+	}
+
+	// convert the object to the external version before sending it to the webhook
+	var versioned versionedObjects
+	gvr := attr.GetResource()
+	if attr.GetOldObject() != nil {
+		versioned.oldObject, err = a.convertor.ConvertToVersion(attr.GetOldObject(), schema.GroupVersion{Group: gvr.Group, Version: gvr.Version})
```

Comment:
Will this break if we rename a type?

Reply:
Reading the [scheme.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go), I think if we register the type with both the new and the old name (via `AddKnownTypeWithName`), the conversion will work.

Reply:
If the webhook is called for the path "apis/indecisive.com/v1/fooname", and realfooname is the internal name, might we end up with "apis/indecisive.com/v1/realfooname" after this call?

Reply:
Changed to use attr.GetKind() to find create a new object, and convert to that object directly.